react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { activeIndex: 0 };
+    this.revolver = this.revolver.bind(this);
+    this.counter = this.counter.bind(this);
+    this.componentDidMount = this.componentDidMount.bind(this);
+    this.prevImg = this.prevImg.bind(this);
+  }
+
+  revolver() {
+    this.intervalId = setInterval(this.counter, 3000);
+  }
+
+  counter() {
+    this.setState({ activeIndex: this.state.activeIndex + 1 });
+    if (this.state.activeIndex === 4) {
+      this.setState({ activeIndex: 0 });
+    }
+  }
+
+  prevImg() {
+    this.setState({ activeIndex: this.state.activeIndex - 1 });
+    if (this.state.activeIndex <= 0) {
+      this.setState({ activeIndex: 4 });
```

Comment:
What if you are given an array with more that 5 images in it, or less? How can we make this work for any array size?